watertap/tools/parameter_sweep.py

```diff
@@ -582,110 +643,95 @@ def _do_param_sweep(model, sweep_params, outputs, local_values, optimize_functio
 
         local_solve_successful_list.append(run_successful)
 
     local_output_dict["solve_successful"] = local_solve_successful_list
 
-    return local_results, local_output_dict
+    return local_output_dict
 
 # ================================================================
 
-def _aggregate_local_results(global_values, local_results, local_output_dict,
-        num_samples, local_num_cases, comm, rank, num_procs):
+def _aggregate_local_results(global_values, local_output_dict, num_samples, local_num_cases,
+        comm, rank, num_procs):
+
+    # Create the dictionary
+    global_results_dict = _create_global_output(local_output_dict, num_samples, comm, rank, num_procs)
 
-    global_results = _aggregate_results(local_results, global_values, comm, num_procs)
-    global_output_dict = _create_global_output(local_output_dict, num_samples, comm, rank, num_procs)
+    # Create the array
+    num_global_samples = np.shape(global_values)[0]
+    global_results_arr = _aggregate_results_arr(global_results_dict, num_global_samples, comm, rank, num_procs)
 
-    return global_results, global_output_dict
+    return global_results_dict, global_results_arr
 
 # ================================================================
 
-def _save_results(sweep_params, outputs, local_values, global_values, local_results,
-        global_results, global_output_dict, csv_results_file, h5_results_file,
-        debugging_data_dir, comm, rank, num_procs, interpolate_nan_outputs):
+def _save_results(sweep_params, local_values, global_values, local_results_dict, global_results_dict,
+        global_results_arr, results_file_name, write_csv, write_h5, debugging_data_dir, comm, rank,
+        num_procs, interpolate_nan_outputs):
 
-    # Make a directory for saved outputs
-    if rank == 0:
-        if csv_results_file is not None:
-            if not csv_results_file.endswith(".csv"):
-                csv_results_file += ".csv"
-            dirname = os.path.dirname(csv_results_file)
-            if dirname != '':
+    if results_file_name is not None:
+        dirname, fname_no_ext, extension = _process_results_filename(results_file_name)
+        if rank == 0:
+            # Create the directories for the results and/or for debugging
+            if extension == '.h5' and write_h5 is False:
+                warnings.warn("An H5 results filename was provided. Outputs will be created", UserWarning)
+                write_h5 = True
+            elif extension == '.csv' and write_csv is False:
+                warnings.warn("A CSV results filename was provided. Outputs will be created", UserWarning)
+                write_csv = True
+
+            if (write_h5 or write_csv) and dirname != '':
                 os.makedirs(dirname, exist_ok=True)
+            # elif not write_h5 and not write_csv:
+            #     warnings.warn("A results filename was provided but neither options to write H5 or csv was selected. No file will be written.", UserWarning)
```

Comment:
This code should either be removed or un-commented -- probably un-commented, given the warnings above.
```suggestion
            elif not write_h5 and not write_csv:
                warnings.warn("A results filename was provided but neither options to write H5 or csv was selected. No file will be written.", UserWarning)
```